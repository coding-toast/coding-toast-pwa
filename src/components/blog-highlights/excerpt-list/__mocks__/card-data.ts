export const query = [
  {
    blogId: '968769ef-231d-5501-8942-ee51d8d1324d',
    title: 'blog-post-placerholder1',
    publishDate: '`January 1, 1970`',
    description: 'first blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  },
  {
    blogId: '1a172731-99f4-5f68-8ead-23324a44f70c',
    title: 'blog-post-placerholder3',
    publishDate: '`January 1, 1970`',
    description: 'third blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  },
  {
    blogId: '011af6f6-6c3f-5eab-81ad-3f73e427dbd8',
    title: 'blog-post-placerholder4',
    publishDate: '`January 1, 1970`',
    description: 'fourth blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  },
  {
    blogId: 'd28f2392-be5f-5fc1-96fb-b3c69e7dce6b',
    title: 'blog-post-placerholder2',
    publishDate: '`January 1, 1970`',
    description: 'second blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  }
];
