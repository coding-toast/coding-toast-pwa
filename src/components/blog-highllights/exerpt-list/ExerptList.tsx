import * as React from 'react'
import { CardBody, CardTitle, CardDesctiption, CardMeta } from '../../card'

export const ExerptList: React.FC = () => {
  return (
    <section>
      <CardBody>
        <CardTitle
          title="Lorem Ipsum"
          backgroundUrl="https://images.unsplash.com/profile-1441298803695-accd94000cac?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64&s=5a9dc749c43ce5bd60870b129a40902f"
        />
        <CardDesctiption
          text="Lorem ipsum dolor sit amet, consectetur adipiscing elit ut aliquam, purus sit amet luctus venenatis,
                            lectus magna fringilla urna, porttitor rhoncus dolor purus non enim praesent elementum facilisis leo,
                            vel fringilla est ullamcorper eget nulla facilisi etiam dignissim diam quis enim lobortis scelerisque
                            fermentum dui faucibus in ornare quam viverra"
        />
        <CardMeta publishDate="November 11, 2016" />
      </CardBody>
      <CardBody>
        <CardTitle
          title="Lorem Ipsum"
          backgroundUrl="https://images.unsplash.com/profile-1441298803695-accd94000cac?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64&s=5a9dc749c43ce5bd60870b129a40902f"
        />
        <CardDesctiption
          text="Lorem ipsum dolor sit amet, consectetur adipiscing elit ut aliquam, purus sit amet luctus venenatis,
                            lectus magna fringilla urna, porttitor rhoncus dolor purus non enim praesent elementum facilisis leo,
                            vel fringilla est ullamcorper eget nulla facilisi etiam dignissim diam quis enim lobortis scelerisque
                            fermentum dui faucibus in ornare quam viverra"
        />
        <CardMeta publishDate="November 11, 2016" />
      </CardBody>
      <CardBody>
        <CardTitle
          title="Lorem Ipsum"
          backgroundUrl="https://images.unsplash.com/profile-1441298803695-accd94000cac?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64&s=5a9dc749c43ce5bd60870b129a40902f"
        />
        <CardDesctiption
          text="Lorem ipsum dolor sit amet, consectetur adipiscing elit ut aliquam, purus sit amet luctus venenatis,
                            lectus magna fringilla urna, porttitor rhoncus dolor purus non enim praesent elementum facilisis leo,
                            vel fringilla est ullamcorper eget nulla facilisi etiam dignissim diam quis enim lobortis scelerisque
                            fermentum dui faucibus in ornare quam viverra"
        />
        <CardMeta publishDate="November 11, 2016" />
      </CardBody>
      <CardBody>
        <CardTitle
          title="Lorem Ipsum"
          backgroundUrl="https://images.unsplash.com/profile-1441298803695-accd94000cac?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64&s=5a9dc749c43ce5bd60870b129a40902f"
        />
        <CardDesctiption
          text="Lorem ipsum dolor sit amet, consectetur adipiscing elit ut aliquam, purus sit amet luctus venenatis,
                            lectus magna fringilla urna, porttitor rhoncus dolor purus non enim praesent elementum facilisis leo,
                            vel fringilla est ullamcorper eget nulla facilisi etiam dignissim diam quis enim lobortis scelerisque
                            fermentum dui faucibus in ornare quam viverra"
        />
        <CardMeta publishDate="November 11, 2016" />
      </CardBody>
    </section>
  )
}
