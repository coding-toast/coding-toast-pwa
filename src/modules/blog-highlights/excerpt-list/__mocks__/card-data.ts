export const query = {
  allStrapiBlogs: {
    edges: [
      {
        node: {
          blogId: '968769ef-231d-5501-8942-ee51d8d1324d',
          title: 'blog-post-placerholder1',
          publishDate: '`January 1, 1970`',
          description: 'first blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: '68e1f043-9989-584f-80e7-2bb2eed3f985',
          title: 'blog-post-placerholder5',
          publishDate: '`January 1, 1970`',
          description: 'fifth blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: '1a172731-99f4-5f68-8ead-23324a44f70c',
          title: 'blog-post-placerholder3',
          publishDate: '`January 1, 1970`',
          description: 'third blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: '011af6f6-6c3f-5eab-81ad-3f73e427dbd8',
          title: 'blog-post-placerholder4',
          publishDate: '`January 1, 1970`',
          description: 'fourth blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: 'd28f2392-be5f-5fc1-96fb-b3c69e7dce6b',
          title: 'blog-post-placerholder2',
          publishDate: '`January 1, 1970`',
          description: 'second blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      }
    ]
  }
}
