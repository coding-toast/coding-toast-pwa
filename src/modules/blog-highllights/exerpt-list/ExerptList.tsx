import * as React from 'react'
import { CardBody, CardTitle, CardDesctiption, CardMeta } from '../../../components/card'
import ExerptListStyles from './ExerptList.module.scss'

interface ExcerptListProps {
  allMarkdownRemark: {
    allMarkdownRemark: {
      edges: Post[]
    }
  }
}
interface Post {
  node: {
    frontmatter: {
      author: string
      banner: string
      description: string
      title: string
      publishDate: string
    }
  }
}
const ExerptList: React.FC<ExcerptListProps> = props => {
  const { allMarkdownRemark } = props
  return (
    <section className={ExerptListStyles.list}>
      {allMarkdownRemark.allMarkdownRemark.edges.map((post: Post) => {
        const { title, banner, description, publishDate } = post.node.frontmatter
        return (
          <CardBody>
            <CardTitle title={title} backgroundUrl={banner} />
            <CardDesctiption text={description} />
            <CardMeta publishDate={publishDate} />
          </CardBody>
        )
      })}
    </section>
  )
}
export default ExerptList
