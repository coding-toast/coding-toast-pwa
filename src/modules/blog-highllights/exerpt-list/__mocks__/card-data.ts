export const query = {
  allMarkdownRemark: {
    edges: [
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'second blog post',
            title: 'blog-post-placerholder2',
            publishDate: 'June 16, 2007'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'third blog post',
            title: 'blog-post-placerholder3',
            publishDate: 'January 30, 2001'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'fourth blog post',
            title: 'blog-post-placerholder4',
            publishDate: 'February 13, 1995'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'fifth blog post',
            title: 'blog-post-placerholder5',
            publishDate: 'December 7, 2005'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'first blog post',
            title: 'blog-post-placerholder1',
            publishDate: 'December 29, 2015'
          }
        }
      }
    ]
  }
}
