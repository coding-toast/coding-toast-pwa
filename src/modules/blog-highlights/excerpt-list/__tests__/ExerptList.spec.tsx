import React from 'react'
import renderer from 'react-test-renderer'
import ExerptList from '../ExcerptList'
import { query } from '../__mocks__/card-data'

it('renders correctly', () => {
  const tree = renderer.create(<ExerptList allMarkdownRemark={query} />).toJSON()
  expect(tree).toMatchSnapshot()
})
