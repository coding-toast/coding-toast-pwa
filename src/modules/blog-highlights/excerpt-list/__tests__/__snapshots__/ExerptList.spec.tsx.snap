// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Row
  as="section"
  md={2}
  noGutters={false}
  xs={1}
>
  <Col>
    <BasicCard
      banner="https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg"
      description="first blog post"
      id="968769ef-231d-5501-8942-ee51d8d1324d"
      publishDate="\`January 1, 1970\`"
      title="blog-post-placerholder1"
    />
  </Col>
  <Col>
    <BasicCard
      banner="https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg"
      description="fifth blog post"
      id="68e1f043-9989-584f-80e7-2bb2eed3f985"
      publishDate="\`January 1, 1970\`"
      title="blog-post-placerholder5"
    />
  </Col>
  <Col>
    <BasicCard
      banner="https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg"
      description="third blog post"
      id="1a172731-99f4-5f68-8ead-23324a44f70c"
      publishDate="\`January 1, 1970\`"
      title="blog-post-placerholder3"
    />
  </Col>
  <Col>
    <BasicCard
      banner="https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg"
      description="fourth blog post"
      id="011af6f6-6c3f-5eab-81ad-3f73e427dbd8"
      publishDate="\`January 1, 1970\`"
      title="blog-post-placerholder4"
    />
  </Col>
  <Col>
    <BasicCard
      banner="https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg"
      description="second blog post"
      id="d28f2392-be5f-5fc1-96fb-b3c69e7dce6b"
      publishDate="\`January 1, 1970\`"
      title="blog-post-placerholder2"
    />
  </Col>
</Row>
`;
